### PR TITLE
fix(proxy) removal of proxy-authorization header

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Clear proxy authorization header only when it is part of original request headers.
Removes it also when the plugin specifies it with exactly same value as in original
request headers.

### CODE REVIEW

**Review by dndx — APPROVED** [review]

File: kong/runloop/handler.lua [hunk 1]
```diff
@@ -1114,10 +1114,9 @@ return {
       local server_port = var.server_port
       ctx.host_port = HOST_PORTS[server_port] or server_port
 
-      -- special handling for proxy-authorization and te headers in case
+      -- special handling for proxy-authorization header in case
       -- the plugin(s) want to specify them (store the original)
       ctx.http_proxy_authorization = var.http_proxy_authorization
-      ctx.http_te                  = var.http_te
```
Comment (dndx) [inline kong/runloop/handler.lua:1120]: Quick question. Is this change related to the `proxy-authorization` change?

Comment (bungle) [reply kong/runloop/handler.lua:1120]: this commit removed the need for context value:
https://github.com/Kong/kong/commit/7e669f3ce55a5e6038c4e254029ef2ea061c4209#diff-093344a0609553c69b475fb5cbd57a202796383eb6fca553d26ac6a1f40d4492L1310

But forgot to remove the context value.
